indico/modules/events/contributions/controllers/display.py

```diff
@@ -0,0 +1,165 @@
+                                                      serialize_contribution, ContributionDisplayReporter)
+from indico.modules.events.contributions.views import WPMyContributions, WPContributions
+from indico.modules.events.util import get_base_ical_parameters
+from indico.web.flask.util import send_file, jsonify_data
+from indico.web.http_api.metadata.serializer import Serializer
+from MaKaC.PDFinterface.conference import ContribToPDF, ContribsToPDF
+from MaKaC.webinterface.rh.base import RH
+from MaKaC.webinterface.rh.conferenceDisplay import RHConferenceBaseDisplay
+
+
+class RHContributionDisplayBase(RHConferenceBaseDisplay):
+
+    CSRF_ENABLED = True
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.contrib
+        }
+    }
+
+    def _checkParams(self, params):
+        RHConferenceBaseDisplay._checkParams(self, params)
+        self.contrib = Contribution.get_one(request.view_args['contrib_id'], is_deleted=False)
+
+
+class RHMyContributions(RHConferenceBaseDisplay):
+    """Display list of current user contributions"""
+
+    def _process(self):
+        contributions = get_contributions_with_user_as_submitter(self.event_new, session.user)
+        return WPMyContributions.render_template('display/user_contribution_list.html', self._conf,
+                                                 event=self.event_new, contributions=contributions)
+
+
+class RHContributionList(RHConferenceBaseDisplay):
+    """Display list of event contributions"""
+
+    def _checkParams(self, params):
+        RHConferenceBaseDisplay._checkParams(self, params)
+        self.contribs = self.event_new.contributions
+        self.reporter = ContributionDisplayReporter(event=self.event_new)
+
+    def _process(self):
+        return WPContributions.render_template('display/contribution_list.html', self._conf, event=self.event_new,
+                                               **self.reporter.get_contrib_report_kwargs())
+
+
+class RHContributionDisplay(RHContributionDisplayBase):
+    """Display page with contribution details """
+
+    def _process(self):
+        ical_params = get_base_ical_parameters(session.user, self.event_new, 'contributions')
+        return WPContributions.render_template('display/contribution_display.html', self._conf,
+                                               contribution=self.contrib, event=self.event_new, **ical_params)
+
+
+class RHContributionAuthor(RHContributionDisplayBase):
+    """Display info about an author"""
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.author
+        }
+    }
+
+    def _checkParams(self, params):
+        RHContributionDisplayBase._checkParams(self, params)
+        self.author = (ContributionPersonLink
+                       .find(contribution=self.contrib, person_id=request.view_args['person_id'])
+                       .filter(~ContributionPersonLink.author_type.in_([AuthorType.none]))
+                       .one())
```

Comment:
no need for the `in`, you can just use `ContributionPersonLink.author_type != AuthorType.none`.

And yes, I wouldn't mix `find()` and `filter()` just like I wouldn't mix kwargs and args in the `find()` call (even though it's not too bad here since the model name is really long..)